spec/xcresult_spec.rb

```diff
@@ -21,7 +21,13 @@
   it 'parse test plan summaries' do
     parser = XCResult::Parser.new(path: path)
 
-    expect(parser).to receive(:execute_cmd).with("xcrun xcresulttool get --format json --path #{path} --id #{summary_id}").and_call_original
+    match = `xcrun xcresulttool version`.match(/xcresulttool version (?<version>\d+),/)
```

Comment:
Please do not execute actual command from spec.

It's better to prepare two contexts with legacy version and new version.

```ruby
before do
  # stub `xcrun xcresulttool version` and returns `xcresulttool_version`
end

context 'with below 23_021.0 ' do
  let(:xcresulttool_version) { '23020.0' }

  it 'should not have --legacy' do
  end
end

context 'with 23_0210 above' do
  let(:xcresulttool_version) { '23021.0' }
end
  
  it 'should have --legacy' do
  end
```

Reply:
Thank you for the review - I've adjusted the PR. Let me know if that works with you. Feel free to change the patch if needed